distributed/utils.py

```diff
@@ -202,6 +203,22 @@ def get_ip_interface(ifname):
     raise ValueError("interface %r doesn't have an IPv4 address" % (ifname,))
 
 
+def shielded(func):
+    """
+    Shield decorated method or function from cancellation. Note that the
+    decorated coroutine will immediately scheduled as a task if the decorated
+    function is invoked.
+
+    See also https://docs.python.org/3/library/asyncio-task.html#asyncio.shield
+    """
+
+    @wraps(func)
+    def _(*args, **kwargs):
+        return asyncio.shield(func(*args, **kwargs))
+
+    return _
```

Comment:
This is not absolutely required but with this decorator we can define methods which are always shielded (e.g. close) instead of relying on remembering it on every invocation